src/pyroboplan/planning/rrt.py

```diff
@@ -261,52 +260,62 @@ def plan(self, q_start, q_goal):
             )
         return self.latest_path
 
-    def extend_or_connect(self, parent_node, q_sample):
+    def extend_or_connect(self, tree, parent_node, q_sample):
         """
         Extends a tree towards a sampled node with steps no larger than the maximum connection distance.
 
         Parameters
         ----------
+            tree : `pyroboplan.planning.graph.Graph`
+                The tree to use when performing this operation.
             parent_node : `pyroboplan.planning.graph.Node`
                 The node from which to start extending or connecting towards the sample.
             q_sample : array-like
                 The robot configuration sample to extend or connect towards.
+
+        Return
+        ------
+            `pyroboplan.planning.graph.Node`, optional
+                The latest node that was added to the tree, or `None` if no node was found.
         """
         # If they are the same node there's nothing to do.
         if np.array_equal(parent_node.q, q_sample):
             return None
 
-        q_out = None
-        q_cur = parent_node.q
+        cur_parent_node = parent_node
+        cur_node = None
         while True:
             # Compute the next incremental robot configuration.
             q_extend = extend_robot_state(
```

Comment:
With the `max_connection_dist` set to `np.inf`, this will always connect to the sample if there's a collision free path, so we don't actually trigger the connect condition in the unit test. 

That said, if I set `max_connection_dist=0.05` in the [rrt connect test](https://github.com/sea-bass/pyroboplan/blob/fix-rrt-connect/test/planning/test_rrt.py#L63), then the test fails to find a path. So unfortunately I think there's definitely something funky going on... But probably not related to this PR :(.

Reply:
Ah, nice! While @adlarkin again did the more principled thing by actually unit testing these constituent functions, I'm gonna be lazy and take your suggestion to just use a finite max connection distance.